src/main/webapp/app/shared/components/customTab/custom-tab.tsx

```diff
@@ -73,14 +74,30 @@ export default connect(mapStateToProps)(CustomTab);
 interface ITabItemProps {
   currentTab: string;
   tabInfo: ITabInfo;
+  tabOnClick?: (tabName: string) => void;
 }
 
-const TabItem: React.FC<ITabItemProps> = ({ currentTab, tabInfo }) => {
+const TabItem: React.FC<ITabItemProps> = ({ currentTab, tabInfo, tabOnClick }) => {
   const isCurrentTab: boolean = tabInfo.tabName === currentTab;
   const btnClassName = classnames('tab-item', isCurrentTab ? 'active-tab' : '');
+
+  const handleClick = (tabName: string) => (event: any) => {
+    if (tabOnClick !== undefined) {
+      tabOnClick(tabName);
+    }
+  };
+
   return (
     <AuthorizationChecker {...tabInfo}>
-      <Button key={tabInfo.tabName} id="tab-btn" color="#07ADE1" className={btnClassName} tag={Link} to={tabInfo.tabRoute}>
+      <Button
+        key={tabInfo.tabName}
+        id="tab-btn"
+        color="#07ADE1"
+        className={btnClassName}
+        tag={Link}
+        to={tabInfo.tabRoute}
+        onClick={handleClick(tabInfo.tabName)}
```

Comment:
```
onClick={handleClick(tabInfo.tabName)}
```
by writing `func()`, you are invoking the function immediately, even the user did not click it. 
This is a bug.
reference: https://reactjs.org/docs/faq-functions.html#why-is-my-function-being-called-every-time-the-component-renders

Reply:
resolved in 9a53e28c330eb98c3e37e2eadf5e2a10eb26abe1